### DISCUSSION TITLE
Explanations on “Node pack name must be a valid Python identifier”

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#55] [Minor issue]
After downloading some example node packs, it is still unclear what a “valid Python identifier” is, when adding local node packs.
There could be explanations on the “Load nodes” prompt.

**Comment by KennedyRichard** [comment]
Thank you for your feedback. You are absolutely right, just mentioning "valid Python identifier" is not enough.

As I mentioned before in my reply to your issue, I'm releasing a new version of Nodezator in a week or so and, among the changes, there is an updated description in the manual on how to determine whether a name is a valid Python identifier.

You can use [str.isidentifier](https://docs.python.org/3/library/stdtypes.html#str.isidentifier) to determine that. In summary, a Python identifier is any name that  starts with a letter or underscore and has only letters, digits and underscores, no punctuation nor spaces.

Since I already added this explanation on the manual and it is only a question of time until it is published, I'll consider this minor issue solved already. I'll close it as soon as you confirm or in a few days.

Thanks again for you feedback, however, regardless of it being something that I tended to already. It is things like this that helps making Nodezator better and several improvements in the past were the result of users posting their concerns/constructive criticism like this.

**Reply by GrimPixel** [reply]
Thanks for telling me about this. I guessed that it may be `__init__.py` in the directory. I am closing it now.
